s3transfer/manager.py

```diff
@@ -45,7 +47,8 @@ def __init__(self,
                  max_submission_queue_size=1000,
                  max_io_queue_size=1000,
                  num_download_attempts=5,
-                 max_in_memory_upload_chunks=10):
+                 max_in_memory_upload_chunks=10,
+                 max_in_memory_download_chunks=10):
```

Comment:
It unfortunately does because ``max_in_memory_upload_chunks`` have a difference performance implication than ``max_in_memory_downloads``. 

Currently if I want to combine the two, the two would have to share the ``SlidingWindowSemaphore`` which may be slower than a regular semaphore because the release of the count of a semaphore is dependent on the lowest sequence number in-progress to finish and that may result in threads not being able to make requests until the lowest sequence number to finish.

With the regular semaphore, the sequence number has no affect on the count when the semaphore is released. So no matter what you can guarantee that if the semaphore is completely exhausted it is because there are ``max_in_memory_upload_chunk`` number of threads completing requests, which that guarantee cannot be made for streaming downloads. In the end, the sharing of the``SlidingWindowSemaphore`` could potentially slow down the in memory uploads.

I think I am on the stance that we re-address these parameters before ``s3transfer`` goes GA to see if we can combine the two (as I do think it is a little strange), but for now since it is not publicly exposed and it requires an additional feature to be able to support a ``max_in_memory_chunks`` parameter effectively.

How does that sound? Does that make sense?